platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/SignedStateFileReader.java

```diff
@@ -87,43 +95,73 @@ public static List<SavedStateInfo> getSavedStateFiles(
         checkSignedStatePath(stateFile);
 
         final DeserializedSignedState returnState;
-
-        record StateFileData(MerkleRoot state, Hash hash, SigSet sigSet) {}
-
-        final StateFileData data = deserializeAndDebugOnFailure(
-                () -> new BufferedInputStream(new FileInputStream(stateFile.toFile())),
-                (final MerkleDataInputStream in) -> {
-                    readAndCheckVersion(in);
-
-                    final Path directory = stateFile.getParent();
-
-                    try {
-                        final MerkleRoot state = snapshotStateReader.apply(in, directory);
-                        final Hash hash = in.readSerializable();
+        StateFileData data = readStateFileData(stateFile, snapshotStateReader);
+
+        final StateFileData normalizedData;
+        if (data.sigSet == null) {
```

Comment:
Can this logic be baked into the `readStateFileData` utility method? That method already has a block for when the file version is `SIG_SET_SEPARATE_VERSION`. Wouldn't it make sense to just read in the sig files there?

Reply:
>Can this logic be baked into the readStateFileData utility method? 

This is how I initially implemented it but then I realized that I will not be able to reuse `SignedStateFileReader#readStateFileData` in this case. Currently, the method may be used in the context of saving the signed state and saving isolated Merkle tree without everything else. If sig set logic was baked in here, I'd have to take care of the fact that sig set files could be absent due to the fact that this is a snapshot of a Merkle tree, not a full signed state. 

I think checking for absence in `readStateFileData` would be a bit more confusing than having this logic defined here.

Reply:
I think it's confusing that `readStateFileData` returns a `StateFileData`, which may or may not be complete, depending in the version of the state file. I think it would be better for `readStateFileData` to always return a complete `StateFileData`, regardless of the version of the state file. 

If we wanted `readStateFileData` to not consider signatures, I think we wouldn't want it to return a `StateFileData` object, which is signature aware. 

(just IMO, up to you, as it isn't a huge deal)